core/src/main/scala/org/apache/spark/status/api/v1/api.scala

```diff
@@ -106,7 +106,17 @@ class ExecutorSummary private[spark](
     val blacklistedInStages: Set[Int],
     @JsonSerialize(using = classOf[ExecutorMetricsJsonSerializer])
     @JsonDeserialize(using = classOf[ExecutorMetricsJsonDeserializer])
-    val peakMemoryMetrics: Option[ExecutorMetrics])
+    val peakMemoryMetrics: Option[ExecutorMetrics],
+    val attributes: Map[String, String]) {
+
+  def replaceExecutorLogs(newExecutorLogs: Map[String, String]): ExecutorSummary = {
```

Comment:
At least `private[spark]`. But I'd prefer either:

- put this in the (only) class that's calling it
- make this a case class and use the `copy` method instead.

Reply:
I tried to convert this to case class first and it didn't work as I expected (I guess that was related to Jackson) so gave up. I'll move this to caller. Thanks for the nice suggestion!